test/common/http/conn_manager_impl_test.cc

```diff
@@ -700,8 +751,7 @@ TEST_F(HttpConnectionManagerImplTest, WebSocketPrefixRewrite) {
   EXPECT_CALL(route_config_provider_.route_config_->route_->route_entry_,
               finalizeRequestHeaders(_, _));
   EXPECT_CALL(route_config_provider_.route_config_->route_->route_entry_, autoHostRewrite())
-      .WillOnce(Return(false));
-  // TODO (rshriram) figure out how to test the auto host rewrite. Need handle over headers.
```

Comment:
I don't see any behavior change with the switch from false to true.  How are we testing that this is performing as expected?

Reply:
Ah shoot. Good catch. Is there a way to get a handle over the request headers, so that I can check if the authority value has been properly rewritten?

Reply:
Are the headers passed into finalizeRequestHeaders the same object which is manipulated for autoHostRewrite?  You could try rewriting  finalizeRequestHeaders mock to Invoke a function which stashes a pointer and then check the headers both in the finalizeRequestHeaders for initial host and after autoHostRewrite for the new host.  I'm not sure it'd work - they might be moved/destroyed before you get a chance to inspect but it's worth a shot :-)

Reply:
You create the headers and "decode" them below. This header map is stored inside the conn man and forwarded on as the request headers. Just store a raw pointer to it before you decode and then you can inspect the request headers whenever you like.